internal/datastore/spanner/migrations/zz_migration.0003_add_caveat.go

```diff
@@ -0,0 +1,31 @@
+package migrations
+
+import (
+	"context"
+
+	"google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+const (
+	addCaveatToRelationTuple     = `ALTER TABLE relation_tuple ADD COLUMN caveat BYTES(4096)`
```

Comment:
The `4096` here is smaller than the other datastore migrations; we might need to document this somewhere

Reply:
this is defined in the proto: https://github.com/authzed/spicedb/pull/773/files#diff-daf9cb335c0ae98207245de93841668f2a4bf8025073e2cdf0af2cdf7589a18eR16 but you are right other datastores are not defining the limit 👍🏻 

do you have any thoughts on what a reasonable starting limit for this could look like?

Reply:
I've documented the size limit of each datastore in https://github.com/authzed/spicedb/pull/773/commits/aa8b50e52f054c8a92ac359b1eb92378a8fff72f.
The proto validation plugin should make sure no more than 4KB are allowed, but I'll add a test to validate it